api/db-cdr/generate-cdr/make-bq-public-data.sh

```diff
@@ -78,15 +85,80 @@ else
   bq --project=$PUBLIC_PROJECT mk $PUBLIC_DATASET
 fi
 
-copy_tables=(achilles_analysis achilles_results achilles_results_concept achilles_results_dist concept concept_relationship criteria db_domain domain vocabulary )
+copy_tables=(achilles_analysis achilles_results achilles_results_concept concept concept_relationship criteria db_domain domain vocabulary )
 for t in "${copy_tables[@]}"
 do
   bq --project=$WORKBENCH_PROJECT rm -f $PUBLIC_PROJECT:$PUBLIC_DATASET.$t
   bq --nosync cp $WORKBENCH_PROJECT:$WORKBENCH_DATASET.$t $PUBLIC_PROJECT:$PUBLIC_DATASET.$t
 done
 
-# Todo, Run queries to make counts suitable for public
+# Round counts for public dataset
+# Set any count below min count to BIN_SIZE, round any above to multiple of BIN_SIZE
+
+# Get person count to set prevalence
+q="select count_value from \`${PUBLIC_PROJECT}.${PUBLIC_DATASET}.achilles_results\` a where a.analysis_id = 1"
+person_count=`bq --quiet --project=$PUBLIC_PROJECT query --nouse_legacy_sql "$q" |  tr -dc '0-9'`
+
+# achilles_results
+bq --quiet --project=$PUBLIC_PROJECT query --nouse_legacy_sql \
+"Update  \`$PUBLIC_PROJECT.$PUBLIC_DATASET.achilles_results\`
+set count_value =
+    case when count_value < ${BIN_SIZE}
+        then ${BIN_SIZE}
+    when Mod(count_value, ${BIN_SIZE}) >= ${BIN_SIZE} / 2
+        then count_value + ${BIN_SIZE} - Mod(count_value, ${BIN_SIZE})
+    when Mod(count_value, ${BIN_SIZE}) < ${BIN_SIZE} / 2
+        then count_value -  Mod(count_value, ${BIN_SIZE})
+    end
+where count_value > 0 and Mod(count_value, ${BIN_SIZE}) > 0"
```

Comment:
Why the second part of this and condition? Seems like that will make counts not get binned if they are less than BIN_SIZE?

Reply:
This is just so it doesn't update ones already a multiple of bin size. save update . thought it might run a little faster.

Reply:
OK. IMHO it's not worth the trouble.